### ISSUE TITLE
Only make pre-commit build sphinx docs if documentation has changed

### ISSUE THREAD

**Issue by cortadocodes** [octue/octue-sdk-python#76]
This will make the pre-commit hooks faster and more efficient.

**Comment by thclark** [comment]
Nope, because documentation integrates API docs and doctest which change on code change too.

However, the `build-docs` hook does require work because at present I'm finding it doesn't rebuild on a doc-only change!
